tests/pytest/test_short_read.py

```diff
@@ -0,0 +1,518 @@
+    # Favor a top-level object/array
+    if nesting_level == 0 and nesting_level != max_nesting_level and not random.randint(0, 5):
+        kind = random.randint(6, 7)
+    else:
+        kind = random.randint(1, 7)
+    if kind == 1:
+        # null
+        return None
+    elif kind == 2:
+        # int
+        return int(rand_num(3))
+    elif kind == 3:
+        # float
+        return float(rand_num(3) + '.' + rand_num(2))
+    elif kind == 4:
+        # string
+        return rand_name(6)
+    elif kind == 5:
+        # bool
+        return rand_bool()
+    elif kind == 6:
+        # array
+        arr = []
+        count = 0 if nesting_level == max_nesting_level else random.randint(1, 10)
+        for i in range(0, count):
+            arr.append(random_json_value(nesting_level + 1, max_nesting_level))
+        return arr
+    elif kind == 7:
+        # object
+        obj = {}
+        count = 0 if nesting_level == max_nesting_level else random.randint(1, 10)
+        for i in range(0, count):
+            obj[rand_name(4)] = random_json_value(nesting_level + 1, max_nesting_level)
+        return obj
+    return None
+
+
+def testCreateKeysRdbFile(env):
+    if os.environ.get('CI'):
+        env.skip()
+    create_keys(env, 'rejson_keys_2.0.0.rdb')
+
+
+class Connection(object):
+    def __init__(self, sock, bufsize=4096, underlying_sock=None):
+        self.sock = sock
+        self.sockf = sock.makefile('rwb', bufsize)
+        self.closed = False
+        self.peer_closed = False
+        self.underlying_sock = underlying_sock
+
+    def close(self):
+        if not self.closed:
+            self.closed = True
+            self.sockf.close()
+            self.sock.close()
+            self.sockf = None
+
+    def is_close(self, timeout=2):
+        if self.closed:
+            return True
+        try:
+            with TimeLimit(timeout):
+                return self.read(1) == ''
+        except Exception:
+            return False
+
+    def flush(self):
+        self.sockf.flush()
+
+    def get_address(self):
+        return self.sock.getsockname()[0]
+
+    def get_port(self):
+        return self.sock.getsockname()[1]
+
+    def read(self, bytes):
+        return self.decoder(self.sockf.read(bytes))
+
+    def read(self, size):
+        return self.decoder(self.sockf.read(size))
+
+    def read_at_most(self, bytes, timeout=0.01):
+        self.sock.settimeout(timeout)
+        return self.decoder(self.sock.recv(bytes))
+
+    def send(self, data):
+        self.sockf.write(self.encoder(data))
+        self.sockf.flush()
+
+    def encoder(self, value):
+        return value.encode('utf-8')
+
+    def decoder(self, value):
+        if isinstance(value, bytes):
+            return value.decode('utf-8')
+        else:
+            return value
+
+    def readline(self):
+        return self.decoder(self.sockf.readline())
+
+    def send_bulk_header(self, data_len):
+        self.sockf.write(self.encoder('$%d\r\n' % data_len))
```

Comment:
As a rule, byte strings in Python (as well as in Rust...) should have the `b'...'` prefix. Plain strings are considered Unicode in Python 3, which is not what you want.